kafka-client/src/test/java/dev/responsive/kafka/internal/stores/CommitBufferTest.java

```diff
@@ -577,6 +581,46 @@ public void shouldOnlyFlushWhenIntervalTriggerElapsed() {
     }
   }
 
+  private Optional<Instant> measureFlushTime(
+      final CommitBuffer<Bytes, Integer> buffer,
+      final AtomicReference<Instant> clock,
+      final long flushOffset,
+      final Instant to
+  ) {
+    buffer.put(Bytes.wrap(new byte[]{18}), VALUE, CURRENT_TS);
+    while (clock.get().isBefore(to)) {
+      buffer.flush(flushOffset);
+      if (table.fetchOffset(KAFKA_PARTITION) == flushOffset) {
+        return Optional.of(clock.get());
+      }
+      clock.set(clock.get().plus(Duration.ofSeconds(1)));
+    }
+    return Optional.empty();
+  }
+
+  @Test
+  public void shouldApplyJitterToFlushInterval() {
+    final AtomicReference<Instant> clock = new AtomicReference<>(Instant.now());
```

Comment:
nit: can we put this in given/when/then and write a small comment about exactly what/how it's testing? I need to critically think in order to figure out what this test is doing

Reply:
I'll definitely add a comment. It's a hard test to format as a `given-when-then` since it's testing randomness over many iterations.